src/pages/bubbles/[number].svelte

```diff
@@ -0,0 +1,169 @@
+<div class="circles" data-cy="bubbles">
+    {#each bubbles as bubble, index}
+        <div class="bubble"
+             on:click={ (event) => pop(event, index) }
+             style="{ getStyle(bubble) }"
+             class:popping="{ bubbles[index].popping }"
+             on:animationiteration={ (event) => animationEndHandler(event, index) }
+        >
+            <div class="inner"></div>
+            <div class="particles"></div>
+        </div>
+    {/each}
+</div>
+<script>
+    import { params } from '@roxi/routify'
+
+    const bubbles = getGeneratedBubbles(checkTimes(parseInt($params.number)))
+
+    function getStyle (bubble) {
+      const styles = [
+        `width: ${bubble.size}px`,
+        `left: ${bubble.left}%`,
+        `height: ${bubble.size}px`,
+        `animation-delay: ${bubble.delay}s`
+      ]
+      if (bubble.duration) {
+        styles.push(`animation-duration: ${bubble.duration}s`)
+      }
+      return styles.join(';')
+    }
+
+    function pop (event, index) {
+      if (!bubbles[index].popping) {
+        bubbles[index].popping = true
+        for (let i = 0; i < Math.random() * 10 + 2; i++) {
+          createParticle(event.target.parentElement.getElementsByClassName('particles')[0])
+        }
+      }
+    }
+
+    function createParticle (parent) {
```

Comment:
Do we need this code copypasted from `src/layout/bubbles.svelte`?

Reply:
Changed, take a look and give me feedback about my solution.